### ISSUE TITLE
AbiEncoder.GetABIParamsEncoded missing prefix?

### ISSUE THREAD

**Issue by Trojaner** [Nethereum/Nethereum#777]
## Expected Behavior

abi.encode() should match AbiEncoder.GetABIParamsEncoded()

## Current Behavior

AbiEncoder is missing some sort of prefix, see below for comparison to abi.encode.

## Steps to Reproduce
```cs
public partial class UniswapV2Params : UniSwapV2ParamsBase { }

public class UniSwapV2ParamsBase
{
    [Parameter("address", "router", 1)]
    public virtual string Router { get; set; }

    [Parameter("address[]", "path", 2)]
    public virtual List<string> Path { get; set; }
}

var uniSwapParams = new UniswapV2Params
{
    Router = RouterAddress,
    Path = path.Select(x => x.GetToken().Address).ToList()
};

var abiEncode = new ABIEncode();
var bytes = abiEncode.GetABIParamsEncoded(uniSwapParams).ToList();
// result: 0x000000000000000000000000a5e0829caced8ffdd4de3c43696c57f7d7a678ff000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000040000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a0630000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa841740000000000000000000000007ceb23fd6bc0add59e62ac25578270cff1b9f6190000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a063
```

```cs
struct UniswapV2Params {
    address router;
    address[] path;
}

UniswapV2Params memory expectedParams = UniswapV2Params({
    /* same parameters */
});

console.logBytes(abi.encode(expectedParams));
// result: 0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000a5e0829caced8ffdd4de3c43696c57f7d7a678ff000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000040000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a0630000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa841740000000000000000000000007ceb23fd6bc0add59e62ac25578270cff1b9f6190000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a063
```

## Context (Environment)

I am trying to pass a struct type depending on what exchange platform I am going to use. I use an enum to define what platform I want to use and then decode the above data to the platforms struct. However this causes an exception on hardhat and the transaction reverts. If I add the prefix manually everything works fine.

## Summary
In short I can't get the same result of abi.encode with AbiEncoder. It seems like it is missing some kind of 32 bytes long prefix. abi.encode starts with `0x0000000000000000000000000000000000000000000000000000000000000020` while AbiEncoder is missing it.

**Comment by juanfranblanco** [comment]
This is a struct as opposed to be a collection of params to encode in an object.
So you will need to use ```AbiValue("struct", uniSwapParams);```

**Comment by Trojaner** [comment]
Thank you for the fast response.
Using
```cs
var bytes = abiEncode.GetABIEncoded(new ABIValue("tuple", uniSwapParams)).ToList();
```
fixed the issue.
